flyteplugins/go/tasks/plugins/webapi/agent/client.go

```diff
@@ -97,17 +97,11 @@ func getFinalContext(ctx context.Context, operation string, agent *Deployment) (
 	return context.WithTimeout(ctx, timeout)
 }
 
-func updateAgentRegistry(ctx context.Context, cs *ClientSet) {
-	agentRegistry := make(Registry)
+func getUpdatedAgentRegistry(ctx context.Context, cs *ClientSet) Registry {
```

Comment:
nit:
```suggestion
func getAgentRegistry(ctx context.Context, cs *ClientSet) Registry {
```

Reply:
FIxed, thank you, and I also added tests